pkgs/by-name/as/asciinema_3/package.nix

```diff
@@ -7,64 +7,61 @@
   testers,
 }:
 
-let
-  self = rustPlatform.buildRustPackage {
-    pname = "asciinema";
-    version = "3.0.1";
+rustPlatform.buildRustPackage (finalAttrs: {
+  pname = "asciinema";
+  version = "3.0.1";
 
-    src = fetchFromGitHub {
-      name = "asciinema-source-${self.version}";
-      owner = "asciinema";
-      repo = "asciinema";
-      rev = "v${self.version}";
-      hash = "sha256-jWRq/LeDdCETiOMkWr9EIWztb14IYGCSo05QPw5HZZk=";
-    };
+  src = fetchFromGitHub {
+    name = "asciinema-source-${finalAttrs.version}";
+    owner = "asciinema";
+    repo = "asciinema";
+    rev = "v${finalAttrs.version}";
+    hash = "sha256-jWRq/LeDdCETiOMkWr9EIWztb14IYGCSo05QPw5HZZk=";
+  };
 
-    cargoHash = "sha256-bGhShwH4BxE3O4/B8KSK1o7IXNDBmGuVt4kx5s8W/go=";
+  cargoHash = "sha256-bGhShwH4BxE3O4/B8KSK1o7IXNDBmGuVt4kx5s8W/go=";
 
-    env.ASCIINEMA_GEN_DIR = "gendir";
+  env.ASCIINEMA_GEN_DIR = "gendir";
 
-    nativeCheckInputs = [ python3 ];
-    nativeBuildInputs = [ installShellFiles ];
+  nativeCheckInputs = [ python3 ];
+  nativeBuildInputs = [ installShellFiles ];
 
-    postInstall = ''
-      installManPage gendir/man/*
-      installShellCompletion --cmd asciinema \
-        --bash gendir/completion/asciinema.bash \
-        --fish gendir/completion/asciinema.fish \
-        --zsh gendir/completion/_asciinema
-    '';
+  postInstall = ''
+    installManPage gendir/man/*
+    installShellCompletion --cmd asciinema \
+      --bash gendir/completion/asciinema.bash \
+      --fish gendir/completion/asciinema.fish \
+      --zsh gendir/completion/_asciinema
+  '';
 
-    strictDeps = true;
+  strictDeps = true;
 
-    passthru = {
-      tests.version = testers.testVersion {
-        package = self;
-        command = "asciinema --version";
-      };
+  passthru = {
+    tests.version = testers.testVersion {
```

Comment:
Can use versionCheckHook here, it will run at build time instead of having to run passthru.tests separately.